camelComponent/src/main/java/io/vantiq/extsrc/camel/VantiqProducer.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2022 Vantiq, Inc.
+ *
+ * All rights reserved.
+ *
+ * SPDX: MIT
+ */
+
+package io.vantiq.extsrc.camel;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vantiq.extjsdk.ExtensionServiceMessage;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Exchange;
+import org.apache.camel.ExchangePattern;
+import org.apache.camel.converter.stream.InputStreamCache;
+import org.apache.camel.InvalidPayloadException;
+import org.apache.camel.support.DefaultProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+public class VantiqProducer extends DefaultProducer {
+    private static final Logger LOG = LoggerFactory.getLogger(VantiqProducer.class);
+    private VantiqEndpoint endpoint;
+    ObjectMapper mapper = new ObjectMapper();
+    
+    public VantiqProducer(VantiqEndpoint endpoint) {
+        super(endpoint);
+        this.endpoint = endpoint;
+    }
+    
+    @SuppressWarnings("unchecked")
+    public void process(Exchange exchange) throws Exception {
+        log.debug("VantiqProducer.process() sending message: {}", exchange.getIn().getBody());
```

Comment:
For streams, that may not be true (though an un-rereadable stream will be quite unlikely here since we're going to turn it into JSON).  Anyway restructured to track things for streams differently.